pkg/ingester/instance_limits.go

```diff
@@ -5,14 +5,27 @@
 
 package ingester
 
-import "github.com/pkg/errors"
+import (
+	"flag"
+
+	"github.com/pkg/errors"
+
+	"github.com/grafana/mimir/pkg/util/globalerror"
+)
+
+const (
+	maxIngestionRateFlag        = "ingester.instance-limits.max-ingestion-rate"
+	maxInMemoryTenantsFlag      = "ingester.instance-limits.max-tenants"
+	maxInMemorySeriesFlag       = "ingester.instance-limits.max-series"
+	maxInflightPushRequestsFlag = "ingester.instance-limits.max-inflight-push-requests"
+)
 
 var (
 	// We don't include values in the message to avoid leaking Mimir cluster configuration to users.
-	errMaxSamplesPushRateLimitReached = errors.New("cannot push more samples: ingester's samples push rate limit reached")
-	errMaxUsersLimitReached           = errors.New("cannot create TSDB: ingesters's max tenants limit reached")
-	errMaxSeriesLimitReached          = errors.New("cannot add series: ingesters's max series limit reached")
-	errTooManyInflightPushRequests    = errors.New("cannot push: too many inflight push requests in ingester")
+	errMaxIngestionRateReached    = errors.New(globalerror.IngesterMaxIngestionRate.MessageWithLimitConfig(maxIngestionRateFlag, "the write request has been rejected because the ingester exceeded the samples ingestion rate limit"))
+	errMaxTenantsReached          = errors.New(globalerror.IngesterMaxTenants.MessageWithLimitConfig(maxInMemoryTenantsFlag, "thr write request has been rejected because the ingester exceeded the allowed number of tenants"))
```

Comment:
```suggestion
	errMaxTenantsReached          = errors.New(globalerror.IngesterMaxTenants.MessageWithLimitConfig(maxInMemoryTenantsFlag, "the write request has been rejected because the ingester exceeded the allowed number of tenants"))
```